homeWork5/lesson1/Program.cs

```diff
@@ -0,0 +1,449 @@
+
+///
+Console.WriteLine($"({ rNumbers1.Numerator } / { rNumbers1.Denumenator }) / ({ rNumbers2.Numerator } / { rNumbers2.Denumenator }) = { rNumbers1 / rNumbers2 }");
+Console.WriteLine($"({ rNumbers1.Numerator } / { rNumbers1.Denumenator }) / ({ rNumbers3.Numerator } / { rNumbers3.Denumenator }) = { rNumbers1 / rNumbers3 }");
+Console.WriteLine($"({ rNumbers1.Numerator } / { rNumbers1.Denumenator }) / ({ rNumbers2.Numerator } / { rNumbers2.Denumenator }) / ({ rNumbers3.Numerator } / { rNumbers3.Denumenator }) = { rNumbers1 / rNumbers2 / rNumbers3 }\n"); 
+
+//%
+Console.WriteLine($"({ rNumbers1.Numerator } / { rNumbers1.Denumenator }) % ({ rNumbers2.Numerator } / { rNumbers2.Denumenator }) = { rNumbers1 % rNumbers2 }");
+Console.WriteLine($"({ rNumbers1.Numerator } / { rNumbers1.Denumenator }) % ({ rNumbers3.Numerator } / { rNumbers3.Denumenator }) = { rNumbers1 % rNumbers3 }");
+Console.WriteLine($"({ rNumbers1.Numerator } / { rNumbers1.Denumenator }) % ({ rNumbers2.Numerator } / { rNumbers2.Denumenator }) % ({ rNumbers3.Numerator } / { rNumbers3.Denumenator }) = { rNumbers1 % rNumbers2 % rNumbers3 }\n");
+
+//++
+Console.WriteLine($"({ rNumbers1.Numerator } / { rNumbers1.Denumenator })++ = { rNumbers1++ }");
+Console.WriteLine($"({ rNumbers2.Numerator } / { rNumbers2.Denumenator })++ = { rNumbers2++ }");
+Console.WriteLine($"({ rNumbers3.Numerator } / { rNumbers3.Denumenator })++ = { rNumbers3++ }\n");
+
+//--
+Console.WriteLine($"({ rNumbers1.Numerator } / { rNumbers1.Denumenator })-- = { rNumbers1-- }");
+Console.WriteLine($"({ rNumbers2.Numerator } / { rNumbers2.Denumenator })-- = { rNumbers2-- }");
+Console.WriteLine($"({ rNumbers3.Numerator } / { rNumbers3.Denumenator })-- = { rNumbers3-- }");
+Console.ReadKey();
+
+public class RatinalNumbers
+{
+    public RatinalNumbers(int numerator, int denumerator)
+    {
+        _numerator = numerator;
+        _denumerator = denumerator;
+    }
+
+    private int _numerator;
+    private int _denumerator;
+
+    public int Numerator
+    {
+        get => _numerator;
+        set => _numerator = value;      
+    }
+
+    public int Denumenator
+    {
+        get => _denumerator;
+        set => _denumerator = value;
+    }
+
+/// <summary>
+/// Переопределенный метод Equals() 
+/// </summary>
+/// <param name="obj">object</param>
+/// <returns>bool</returns>
+    public override bool Equals(object? obj)
+    {
+        if (obj is not RatinalNumbers rnumbers)
+        {
+           return false;
+        }
+        return Numerator == rnumbers.Numerator &&
+               Denumenator == rnumbers.Denumenator;
+    }
+
+/// <summary>
+/// Переопределенный метод GetHashCode()
+/// </summary>
+/// <returns>int</returns>
+    public override int GetHashCode()
+    {
+        int hashcode = _numerator.GetHashCode();
+        hashcode = 31 * hashcode + _denumerator.GetHashCode();
+        return hashcode;

```

Comment:
Будет переполнение у Вас.
```suggestion
        unchecked {
        int hashcode = _numerator.GetHashCode();
        hashcode = 31 * hashcode + _denumerator.GetHashCode();
        return hashcode;
        }
```